tests/diagnostics/ctor-noncopyable-retval-legal.slang

```diff
@@ -0,0 +1,103 @@
+//TEST:SIMPLE(filecheck=CHECK):
+
+// Test to catch compilation time errors with return values in constructors within a NonCopyable struct.
+// A constructor from the callsite's point of view is a function that returns the struct type.
+// A constructor from the inside the body is treated as a function that modifies `this` and not return.
+
+// In the following test-cases where the constructor either returns an expr or no value,
+// within a condition for early exit or not, it automatically gets resolved as
+// return *this; or return;
+
+[__NonCopyableType] struct S
```

Comment:
We have opportunity to turn this test into an executable test. If we actually run the shader and write something to a buffer, then getting the correct execution result will imply there are no compile errors.

Reply:
definitely. do you recommend any example tests that I can look up ?